src/ecs/resources/input.rs

```diff
@@ -45,14 +83,30 @@ impl InputHandler {
                             // second `Pressed` event.
                         }
                         Entry::Vacant(entry) => {
-                            entry.insert(KeyQueryState::Queried);
+                            entry.insert(KeyQueryState::NotQueried);
                         }
                     }
                 }
                 Event::KeyboardInput(ElementState::Released, _, Some(key_code)) => {
                     self.pressed_keys.remove(&key_code);
                 }
-                Event::Focused(false) => self.pressed_keys.clear(),
+                Event::MouseInput(ElementState::Pressed, button) => {
+                    match self.pressed_mouse_buttons.entry(button) {
+                        Entry::Occupied(_) => {
+                            // as above
+                        }
+                        Entry::Vacant(entry) => {
```

Comment:
If we are really only using one branch of the `match` statement, couldn't this entire statement be shortened down to:

```rust
if let Entry::Vacant(entry) = self.pressed_mouse_buttons.entry(button) {
    entry.insert(KeyQueryState::NotQueried);
}
```